R/utils.R

```diff
@@ -40,17 +40,9 @@ lengths <- function(x) {
 }
 
 # A 'size' must be an integer greater than 1, returned as a double so we have a larger range
-parse_size <- function(x) {
-  nme <- substitute(x) %||% "NULL"
-
-  if (rlang::is_scalar_integerish(x) && !is.na(x) && !is.infinite(x) && x > 0) {
-    return(as.numeric(x))
-  }
-
-  stop(
-    sprintf("`%s` is not a valid size:\n  Must be a positive integer.", as.character(nme)),
-    call. = FALSE
-  )
+parse_size <- function(x, call = caller_env()) {
```

Comment:
Do you also the `arg` argument so it doesn't always report the problem as the `x` argument?